CMakeLists.txt

```diff
@@ -34,14 +34,12 @@ set(CMAKE_C_STANDARD_REQUIRED true)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_C_STANDARD 11)
 include(GenerateExportHeader)
-set(CMAKE_CXX_FLAGS " -Wall -pedantic -Werror -Wno-deprecated-declarations -D_GLIBCXX_USE_CXX11_ABI=0 -fstack-protector-strong --param=ssp-buffer-size=4 -O3 -D_FORTIFY_SOURCE=2 ${CMAKE_CXX_FLAGS}")
```

Comment:
Seems to me like a good idea to add back `-Wno-deprecated-declarations`, as without it there's a lot of warnings when building (at least on my end), especially with https://doc.qt.io/qt-5/qdir-obsolete.html#operator-eq-1

Reply:
I mean yes and no. It technically should throw warnings if something is deprecated. That's basically the point of deprecation

Reply:
~~meanwhile CurseForge: deprecation = removal~~

Reply:
Yes, but either we remove the things that are giving the warnings right now, or we do it some other time, and use this flag to not spam the console when building. It's 100+ warnings...

Reply:
We mostly can't remove these errors if we want to maintain compatibility with Qt5.12. One option could be to fix these warnings but use QT_VERSION_CHECK

Reply:
It's possible... Actually, looking through it, seems like most of the errors come from the FileSystem header, so maybe there's not that many places that would need this change

Reply:
I think I have all deprecation warnings covered in #575. So I might disable deprecation stuff here, but remove the flag again in Qt 6 PR

Reply:
I think I have all deprecation warnings covered in #575. So I might disable deprecation stuff here, but remove the flag again in Qt 6 PR

Reply:
Fine by me!